app/controllers/delete_logs_controller.rb

```diff
@@ -0,0 +1,196 @@
+    render "logs/delete_logs"
+  end
+
+  def delete_sales_logs_confirmation
+    default_attributes = {
+      current_user:,
+      log_filters: @session_filters,
+      log_type: :sales,
+    }.merge sales_logs_paths
+    attributes = form_attributes.merge(default_attributes)
+    attributes[:selected_ids] = [] unless attributes.key? :selected_ids
+    @delete_logs_form = Forms::DeleteLogsForm.new(attributes)
+    if @delete_logs_form.valid?
+      render "logs/delete_logs_confirmation"
+    else
+      render "logs/delete_logs"
+    end
+  end
+
+  def discard_sales_logs
+    logs = SalesLog.find(params.require(:ids))
+    discard logs
+
+    redirect_to sales_logs_path, notice: I18n.t("notification.logs_deleted", count: logs.count)
+  end
+
+  def delete_lettings_logs_for_organisation
+    @delete_logs_form = delete_logs_form(log_type: :lettings, paths: lettings_logs_for_organisation_paths)
+    render "logs/delete_logs"
+  end
+
+  def delete_lettings_logs_for_organisation_with_selected_ids
+    @delete_logs_form = delete_logs_form(selected_ids:, log_type: :lettings, paths: lettings_logs_for_organisation_paths)
+    render "logs/delete_logs"
+  end
+
+  def delete_lettings_logs_for_organisation_confirmation
+    default_attributes = {
+      current_user:,
+      log_filters: @session_filters,
+      log_type: :lettings,
+    }.merge lettings_logs_for_organisation_paths
+    attributes = form_attributes.merge(default_attributes)
+    attributes[:selected_ids] = [] unless attributes.key? :selected_ids
+    @delete_logs_form = Forms::DeleteLogsForm.new(attributes)
+    if @delete_logs_form.valid?
+      render "logs/delete_logs_confirmation"
+    else
+      render "logs/delete_logs"
+    end
+  end
+
+  def discard_lettings_logs_for_organisation
+    logs = LettingsLog.find(params.require(:ids))
+    discard logs
+
+    redirect_to lettings_logs_organisation_path, notice: I18n.t("notification.logs_deleted", count: logs.count)
+  end
+
+  def delete_sales_logs_for_organisation
+    @delete_logs_form = delete_logs_form(log_type: :sales, paths: sales_logs_for_organisation_paths)
+    render "logs/delete_logs"
+  end
+
+  def delete_sales_logs_for_organisation_with_selected_ids
+    @delete_logs_form = delete_logs_form(selected_ids:, log_type: :sales, paths: sales_logs_for_organisation_paths)
+    render "logs/delete_logs"
+  end
+
+  def delete_sales_logs_for_organisation_confirmation
+    default_attributes = {
+      current_user:,
+      log_filters: @session_filters,
+      log_type: :sales,
+    }.merge sales_logs_for_organisation_paths
+    attributes = form_attributes.merge(default_attributes)
+    attributes[:selected_ids] = [] unless attributes.key? :selected_ids
+    @delete_logs_form = Forms::DeleteLogsForm.new(attributes)
+    if @delete_logs_form.valid?
+      render "logs/delete_logs_confirmation"
+    else
+      render "logs/delete_logs"
+    end
+  end
```

Comment:
As you repeat the first block of this method, would you consider extracting the params out into a separate method?

```ruby
  def delete_sales_logs_for_organisation_confirmation
    @delete_logs_form = Forms::DeleteLogsForm.new(attributes_for_sales_logs(organisation: true))
    if @delete_logs_form.valid?
      render "logs/delete_logs_confirmation"
    else
      render "logs/delete_logs"
    end
  end
  
  private
  
  # ...
  
  def attributes_for_sales_logs(organisation: false)
     attributes = {
      current_user:,
      log_filters: @session_filters,
      log_type: :sales,
    }
    attributes.merge(organisation ? sales_logs_for_organisation_paths : sales_logs_paths)
  end 
```

Reply:
I hadn't fully looked through this suggested change, but I have refactored in a similar but different way.
I've retained the delete_logs_form private method, but made it able to be used by all controller methods wanting a delete logs form. I will push that along with a range of other changes soon